lightningd/peer_control.c

```diff
@@ -776,10 +777,18 @@ static void gossipd_getpeers_complete(struct subd *gossip, const u8 *msg,
 			json_array_end(response);
 		}
 
-		json_add_node_decoration(response, nodes, &p->id);
+		/* Search gossip reply for this ID, to add extra info. */
+		for (size_t i = 0; i < tal_len(nodes); i++) {
+			if (pubkey_eq(&nodes[i]->nodeid, &p->id)) {
```

Comment:
I'm getting an invalid pointer here, so I crash in listpeers now. will poke around to try and figure out why...